File: src/main/java/org/noxet/noxetserver/util/FancyTimeConverter.java
package org.noxet.noxetserver.util;

public class FancyTimeConverter {
    public static String deltaSecondsToFancyTime(int seconds) {
        return deltaSecondsToFancyTime(seconds, false);
    }

    public static String deltaSecondsToFancyTime(int seconds, boolean secondSpecific) {
        int days = seconds / 86400,
            hours = seconds / 3600 % 24,
            minutes = seconds / 60 % 60;
        seconds %= 60;

        StringBuilder stringBuilder = new StringBuilder();

        if(days > 0)
            stringBuilder.append(days).append("d ");

        if(hours > 0)
            stringBuilder.append(hours).append("h ");

        if(days == 0 && minutes > 0)
            stringBuilder.append(minutes).append("m ");

        if(days == 0 && hours == 0 && (secondSpecific || minutes == 0) && seconds > 0)
            stringBuilder.append(seconds).append("s ");

        stringBuilder.deleteCharAt(stringBuilder.length() - 1);

        return stringBuilder.toString();
    }
}

File: src/main/java/org/noxet/noxetserver/util/FancyTimeConverter.scala
package org.noxet.noxetserver.util

import scala.collection.mutable

object FancyTimeConverter:
    def deltaSecondsToFancyTime(seconds: Int, secondSpecific: Boolean = false): String =
        val days = seconds / 86400
        val hours = seconds / 3600 % 24
        val minutes = seconds / 60 % 60

        val parts = mutable.ListBuffer[String]()

        if days > 0 then
            parts += days + "d"

        if hours > 0 then
            parts += hours + "h"

        if days == 0 && minutes > 0 then
            parts += minutes + "m"

        if days == 0 && hours == 0 && (secondSpecific || minutes == 0) then
            parts += seconds % 60 + "s"

        parts.mkString(" ")